### PR TITLE
Hide Azure.Core diagnostic types

### LINKED ISSUE #7984: Make ClientDiagnostics and DiagnosticScope internal shared by source

### PR DESCRIPTION
**Pull request by pakrym**
Fixes: https://github.com/Azure/azure-sdk-for-net/issues/7984

We don't want `DiagnosticScope` and `ClientDiagnostics` to be public yet. Share them as source.

# Breaking change

`ClientDiagnostics` and `DiagnosticsScope` are included as source now and clients have to create own instances of `ClientDiagnostics`.

``` C#
_clientDiagnostics = new ClientDiagnostics(options);
```

### CODE REVIEW

**Comment by pakrym** [comment]
cc @lmolkova, this also changes DiagnosticSource names to be client namespaces to allow future filtering. Filed https://github.com/Azure/azure-sdk-for-net/issues/7982 to react in OpenTelemetry.

File: sdk/storage/Azure.Storage.Common/swagger/Generator/src/generator.ts [hunk 1]
```diff
@@ -169,11 +170,13 @@ function generateOperation(w: IndentWriter, serviceModel: IServiceModel, group:
     if (sync) {
         w.line(`/// <param name="async">Whether to invoke the operation asynchronously.  The default value is true.</param>`);
     }
+    w.line(`/// <param name="${clientDiagnostics}">The ClientDiagnostics instance used for operation reporting.</param>`);
     w.line(`/// <param name="${operationName}">Operation name.</param>`);
     w.line(`/// <param name="${cancellationName}">Cancellation token.</param>`);
     w.line(`/// <returns>${operation.response.model.description || returnType.replace(/</g, '{').replace(/>/g, '}')}</returns>`);
     w.write(`public static async System.Threading.Tasks.ValueTask<${sendMethodReturnType}> ${methodName}(`);
     w.scope(() => {
+        w.line(`Azure.Core.Pipeline.ClientDiagnostics ${clientDiagnostics},`);
```
Comment (pakrym) [inline sdk/storage/Azure.Storage.Common/swagger/Generator/src/generator.ts]: @tg-msft I hate to make it the first parameter but the way pipeline parameter is done I wasn't able to inject a second one without messing up other logic.

**Review by heaths — APPROVED** [review]
Signing off on Azure.Core and Key Vault changes.

**Review by jsquire — APPROVED** [review]
Event Hubs bits look good to me.

**Review by annelo-msft — APPROVED** [review]
Good from AppConfig perspective.

**Review by lmolkova — APPROVED** [review]

**Review by tg-msft — APPROVED** [review]
Thanks for doing this!

**Pull request merged**